src/test/kotlin/org/rust/ide/intentions/DemorgansLawIntentionTest.kt

```diff
@@ -0,0 +1,62 @@
+package org.rust.ide.intentions
+
+import org.rust.lang.RustTestCaseBase
+
+class DemorgansLawIntentionTest : RustTestCaseBase() {
+    override val dataPath = "org/rust/ide/intentions/fixtures/demorgan_law/"
+
+    fun testOr() = checkByFile {
+        openFileInEditor("or.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
+
+    fun testOrNot() = checkByFile {
+        openFileInEditor("or_not.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
+
+    fun testNotOr() = checkByFile {
+        openFileInEditor("not_or.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
+
+    fun testComplex1() = checkByFile {
+        openFileInEditor("complex1.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
+
+    fun testComplex2() = checkByFile {
+        openFileInEditor("complex2.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
+
+    fun testComplex3() = checkByFile {
+        openFileInEditor("complex3.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
+
+    fun testComplex4() = checkByFile {
+        openFileInEditor("complex4.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
+
+    fun testComplex5() = checkByFile {
+        openFileInEditor("complex5.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
+
+    fun testComplex6() = checkByFile {
+        openFileInEditor("complex6.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
+
+    fun testComplex7() = checkByFile {
+        openFileInEditor("complex7.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
+
+    fun testComplex8() = checkByFile {
+        openFileInEditor("complex8.rs")
+        myFixture.launchAction(DemorgansLawIntention())
+    }
```

Comment:
i check reversible in seperated fixtures, and since Kotlin doesn't have a Macro system to generate boilerplate code i will find another way to eliminate the repetitivity.